turbinia/processors/partitions.py

```diff
@@ -87,18 +87,29 @@ def GetPathSpecByLocation(path_specs, location):
   Returns:
     dfVFS.path_spec for the given location or None if not found.
   """
+  path_spec_chain = []
   for path_spec in path_specs:
     child_path_spec = path_spec
+    path_spec_chain.insert(0, child_path_spec)
     fs_location = getattr(path_spec, 'location', None)
     if fs_location and fs_location == location:
+      log.debug(
+          'Found path_spec {0!s} for location {1!s} in list {2!s}'.format(
+              child_path_spec, fs_location, path_spec_chain))
       return child_path_spec
     while path_spec.HasParent():
       type_indicator = path_spec.type_indicator
       if type_indicator in dfvfs_definitions.VOLUME_SYSTEM_TYPE_INDICATORS:
         fs_location = getattr(path_spec, 'location', None)
         break
       path_spec = path_spec.parent
+      path_spec_chain.insert(0, path_spec)
     if fs_location == location:
+      log.debug(
+          'Found path_spec {0!s} for location {1!s} in list {2!s}'.format(
+              child_path_spec, fs_location, path_spec_chain))
```

Comment:
As above.

Reply:
fixed

Reply:
Still printing the `child_path_spec` object here. Could also copy to dict, but maybe not needed since it'll be in the list anyway?

Reply:
Oh, right, thanks.  Fixed here and above.